### PR TITLE
Supported file formats documentation reworked

### PR DESCRIPTION
**Pull request by comradekingu**
Ready but not ready.

### CODE REVIEW

File: docs/formats.rst [hunk 1]
```diff
@@ -101,31 +101,31 @@ Below are listed capabilities of all supported formats.
 
 .. [#m] See :ref:`bimono`
 .. [#p] Plurals are necessary to properly localize strings with variable count.
-.. [#n] Comments can be used to pass additional information about string to translate.
-.. [#c] Context is used to differentiate same strings used in different scope (for example `Sun` can be used as abbreviated name of day or as a name of our closest star).
-.. [#l] Location of string in source code might help skilled translators to figure out how the string is used.
-.. [#a] Additional states supported by the file format in addition to not translated and translated.
-.. [#x] XML comment placed before the ``<string>`` element is parsed as a developer comment.
+.. [#n] Comments can be used to pass additional info about string to translate.
```
Comment (comradekingu) [inline docs/formats.rst]: About ? string to translate.
what, the?

File: docs/formats.rst [hunk 2]
```diff
@@ -101,31 +101,31 @@ Below are listed capabilities of all supported formats.
...
+.. [#c] Context is used to differentiate identical strings used in different scopes (for example `Sun` can be used as an abbreviated name of the day "Sunday" or as the name of our closest star).
+.. [#l] Location of a string in source code might help proficient translators figure out how the string is used.
+.. [#a] Additional states supported by the file format in addition to "Not translated" and "Translated".
+.. [#x] XML comment placed before the ``<string>`` element, parsed as a developer comment.
 .. [#f] See :ref:`custom-checks`
-.. [#po] The Gettext type comments are used as flags.
-.. [#xl] The flags are extracted from non standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through ``maxwidth`` attribute as defined in the Xliff standard, see :ref:`xliff-flags`.
+.. [#po] The gettext type comments are used as flags.
+.. [#xl] The flags are extracted from the non-standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through the ``maxwidth`` attribute as defined in the XLIFF standard, see :ref:`xliff-flags`.
```
Comment (comradekingu) [inline docs/formats.rst]: attributes?

File: docs/formats.rst [hunk 3]
```diff
@@ -231,36 +231,36 @@ is one of many standards in this area.
...
 .. versionchanged:: 3.3
 
-   Weblate did ignore the state attribute prior to the 3.3 release.
+   Weblate ignored the state attribute prior to the 3.3 release.
 
-The ``state`` attribute in the file is partially processed and mapped to needs
-edit state in Weblate (the following states are used to flag the string as
-needing edit if there is some target present: ``new``, ``needs-translation``,
+The ``state`` attribute in the file is partially processed and mapped to the
+"Needs edit" state in Weblate (the following states are used to flag the string as
```
Comment (comradekingu) [inline docs/formats.rst]: I feel like this has been changed to "Marked for edit" in itself not ideal, but I can't find out what "Needs edit" is now.

File: docs/formats.rst [hunk 4]
```diff
@@ -231,36 +231,36 @@ is one of many standards in this area.
...
 ``needs-adaptation``, ``needs-l10n``). Should the ``state`` attribute be
-missing a string is considered translated as soon as a ``<target>`` element
+missing, a string is considered translated as soon as a ``<target>`` element
 exists.
 
-Also if the translation string has ``approved="yes"`` it will be imported into Weblate
-as approved, anything else will be imported as waiting for review (which matches XLIFF
-specification).
+If the translation string has ``approved="yes"``, it will also be imported into Weblate
+as "Approved", anything else will be imported as "Waiting for review" (which matches the
```
Comment (comradekingu) [inline docs/formats.rst]: Is "Approved" correct?

File: docs/formats.rst [hunk 5]
```diff
@@ -231,36 +231,36 @@ is one of many standards in this area.
...
+XLIFF specification).
 
-That means that when using XLIFF format, it is strongly recommended to enable Weblate
-review process, in order to see and change the approved state of strings.
+That means that when using the XLIFF format, it is strongly recommended to turn on the
+Weblate review process, in order to see and change the approved state of strings.
 See :ref:`reviews`.
 
-Similarly on importing such files, you should choose
+Similarly upon importing such files, you should choose
```
Comment (comradekingu) [inline docs/formats.rst]: Where is this chosen?

File: docs/formats.rst [hunk 6]
```diff
@@ -295,13 +295,13 @@ attributes from the Xliff specification:
 The ``font`` attribute is parsed for font family, size and weight, the above
 example shows all of that, though only font family is required. Any whitespace
 in the font family is converted to underscore, so ``Source Sans Pro`` becomes
-``Source_Sans_Pro``, please keep that in mind when naming font group (see
+``Source_Sans_Pro``, please keep that in mind when naming the font group (see
```
Comment (comradekingu) [inline docs/formats.rst]: s/naming/picking/ ?

File: docs/formats.rst [hunk 7]
```diff
@@ -648,20 +648,20 @@ JSON i18next files
...
-with English strings.
+i18next translations are monolingual, so it is recommended to specify a base file
+with (what is most often the) English strings.
 
 .. note::
 
-   Weblate supports i18next JSON v3 format. The v2 and v1 variants are mostly
-   compatible, with exception of handling plurals.
+   Weblate supports the i18next JSON v3 format. The v2 and v1 variants are mostly
+   compatible, with the exception of handling plurals.
```
Comment (comradekingu) [inline docs/formats.rst]: exception of not handling ### ~they are not handled
OR
exception of how plurals are handled ### ~they are handled differently

File: docs/formats.rst [hunk 8]
```diff
@@ -735,13 +735,13 @@ Example file:
 .. versionadded:: 2.3
 
 .NET Resource (.resx) file is a monolingual XML file format used in Microsoft
```
Comment (comradekingu) [inline docs/formats.rst]: Not sure here. Only thing I could find was "VB.NET resource"… ".NET Applications" also seems iffy.

File: docs/formats.rst [hunk 9]
```diff
@@ -735,13 +735,13 @@ Example file:
...
-.NET Applications. It works with .resw files as well as they use identical
+.NET applications. It works with .resw files as well as when they use identical
```
Comment (comradekingu) [inline docs/formats.rst]: What works?

File: docs/formats.rst [hunk 10]
```diff
@@ -769,18 +769,18 @@ CSV files
...
 
-* Files with header defining fields (source, translation, location, ...). This
-  is recommended approach as it's least error prone.
-* Files with two fields - source and translation (in this order), choose
+* Files with header defining fields (source, translation, location, …). This
+  is the recommended approach, as it is the least error prone.
+* Files with two fields—source and translation (in this order), choose
   :guilabel:`Simple CSV file` as file format
 * Files with fields as defined by translate-toolkit: location, source,
   target, id, fuzzy, context, translator_comments, developer_comments
```
Comment (comradekingu) [inline docs/formats.rst]: ?
```suggestion
  target, ID, fuzzy, context, translator_comments, developer_comments
```

File: docs/formats.rst [hunk 11]
```diff
@@ -769,18 +769,18 @@ CSV files
...
 
 .. warning::
 
-   The CSV format currently automatically detects dialect of the CSV file. In
-   some cases the automatic detection might fail and you will get mixed
-   results. This is especially true for the CSV files with newlines in the
+   The CSV format currently automatically detects the dialect of the CSV file.
+   In some cases the automatic detection might fail and you will get mixed
+   results. This is especially true for CSV files with newlines in the
    values. As a workaround it is recommended to avoid omitting quoting characters.
```
Comment (comradekingu) [inline docs/formats.rst]: If not, what is an "omitting quoting" character?
```suggestion
   values. As a workaround it is recommended to omit quoting characters.
```

File: docs/formats.rst [hunk 12]
```diff
@@ -231,36 +231,36 @@ is one of many standards in this area.
 
 XLIFF is usually used as bilingual, but Weblate supports it as monolingual as well.
```
Comment (comradekingu) [inline docs/formats.rst]: There is no link to the XLIFF specification.
other than maybe "attribute as defined in the XLIFF standard, see :ref:`xliff-flags`." earlier.

It should IMO be similar to:
```
.. seealso::

    `Specification of Joomla language files <https://docs.joomla.org/Specification_of_language_files>`_,
```

File: docs/formats.rst (hunk 9 above)
Comment (orangesunny) [reply docs/formats.rst:738]: @nijel

File: docs/formats.rst (hunk 8 above)
Comment (orangesunny) [reply docs/formats.rst:737]: @nijel

File: docs/formats.rst (hunk 2 above)
Comment (nijel) [reply docs/formats.rst:111]: Yes, XML attributes, see http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#maxwidth for specs.

File: docs/formats.rst (hunk 1 above)
Comment (nijel) [inline docs/formats.rst:104]:
```suggestion
.. [#n] Comments can be used to pass additional info about the string to translate.
```

File: docs/formats.rst (hunk 12 above)
Comment (nijel) [reply docs/formats.rst:232]: Good point, the specification is here: http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html

File: docs/formats.rst (hunk 3 above)
Comment (nijel) [reply docs/formats.rst:242]: Not change, see https://github.com/WeblateOrg/weblate/issues/2240

File: docs/formats.rst (hunk 4 above)
Comment (nijel) [reply docs/formats.rst:249]: Yes, this is how this state is currently called:

https://github.com/WeblateOrg/weblate/blob/b19d4bd0834411ec5963a0f835f48730238466e1/weblate/utils/state.py#L33

File: docs/formats.rst (hunk 5 above)
Comment (nijel) [reply docs/formats.rst:256]: I the upload form.

File: docs/formats.rst (hunk 6 above)
Comment (nijel) [reply docs/formats.rst:298]: Naming, when creating it in Weblate.

File: docs/formats.rst (hunk 9 above)
Comment (comradekingu) [reply docs/formats.rst:737]: My question is concerning what "it" is here. I am unsure what works in the first place.

File: docs/formats.rst [hunk 13]
```diff
@@ -227,40 +227,40 @@ XLIFF
...
+XLIFF specification).
 
-That means that when using XLIFF format, it is strongly recommended to enable Weblate
-review process, in order to see and change the approved state of strings.
+That means that when using the XLIFF format, it is strongly recommended to turn on the
+Weblate review process, in order to see and change the approved state of strings.
 See :ref:`reviews`.
 
-Similarly on importing such files, you should choose
+Similarly upon importing such files, you should choose
```
Comment (comradekingu) [inline docs/formats.rst:256]:
```suggestion
Similarly upon importing such files (in the upload form), you should choose
```

File: docs/formats.rst (hunk 7 above)
Comment (nijel) [reply docs/formats.rst]: They are handled differently, see https://www.i18next.com/misc/json-format#i-18-next-json-v3:

> The only difference to v2 are the plural suffixes for languages with multiple plural forms.

File: docs/formats.rst (hunk 8 above)
Comment (nijel) [reply docs/formats.rst:736]: It seems to be called just "XML resource files", see https://docs.microsoft.com/cs-cz/dotnet/framework/resources/working-with-resx-files-programmatically. I've followed translate-toolkit naming here: http://docs.translatehouse.org/projects/translate-toolkit/en/latest/formats/resx.html

File: docs/formats.rst (hunk 9 above)
Comment (nijel) [reply docs/formats.rst:737]: AFAIK the only real difference is extension.

Looking at https://lingohub.com/developers/resource-files/resw-resx-localization:

> These 2 file types have different extensions (.resw, .resx) but the content is the same. The only difference is in the usage of these file types in your application development process.

File: docs/formats.rst [hunk 14]
```diff
@@ -101,31 +101,31 @@ Below are listed capabilities of all supported formats.
...
+.. [#c] Context is used to differentiate identical strings used in different scopes (for example `Sun` can be used as an abbreviated name of the day "Sunday" or as the name of our closest star).
+.. [#l] Location of a string in source code might help proficient translators figure out how the string is used.
+.. [#a] Additional states supported by the file format in addition to "Not translated" and "Translated".
+.. [#x] XML comment placed before the ``<string>`` element, parsed as a developer comment.
 .. [#f] See :ref:`custom-checks`
-.. [#po] The Gettext type comments are used as flags.
-.. [#xl] The flags are extracted from non standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through ``maxwidth`` attribute as defined in the Xliff standard, see :ref:`xliff-flags`.
+.. [#po] The gettext type comments are used as flags.
+.. [#xl] The flags are extracted from the non-standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through the ``maxwidth`` attribute as defined in the XLIFF standard, see :ref:`xliff-flags`.
```
Comment (comradekingu) [inline docs/formats.rst:111]:
```suggestion
.. [#xl] The flags are extracted from the non-standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through the ``maxwidth`` [attribute](http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#maxwidth) as defined in the XLIFF standard, see :ref:`xliff-flags`.
```

File: docs/formats.rst [hunk 15]
```diff
@@ -101,31 +101,31 @@ Below are listed capabilities of all supported formats.
...
+.. [#l] Location of a string in source code might help proficient translators figure out how the string is used.
+.. [#a] Additional states supported by the file format in addition to "Not translated" and "Translated".
+.. [#x] XML comment placed before the ``<string>`` element, parsed as a developer comment.
 .. [#f] See :ref:`custom-checks`
-.. [#po] The Gettext type comments are used as flags.
-.. [#xl] The flags are extracted from non standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through ``maxwidth`` attribute as defined in the Xliff standard, see :ref:`xliff-flags`.
+.. [#po] The gettext type comments are used as flags.
+.. [#xl] The flags are extracted from the non-standard attibute ``weblate-flags`` for all XML based formats. Additionally ``max-length:N`` is supported through the ``maxwidth`` [attribute](http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#maxwidth) as defined in the XLIFF standard, see :ref:`xliff-flags`.
```
Comment (comradekingu) [inline docs/formats.rst:111]: It is possible https://docs.readthedocs.io/en/stable/guides/intersphinx.html is a better idea for this link

File: docs/formats.rst [hunk 16]
```diff
@@ -227,40 +227,40 @@ XLIFF
     pair: XLIFF; file format
 
 XML-based format created to standardize translation files, but in the end it
-is one of many standards in this area.
+is one of `many standards <https://xkcd.com/927/>`_, in this area.
 
 XLIFF is usually used as bilingual, but Weblate supports it as monolingual as well.
```
Comment (comradekingu) [inline docs/formats.rst:232]:
```suggestion
[XLIFF](http://docs.oasis-open.org/xliff/xliff-core/xliff-core.html) is usually used as bilingual, but Weblate supports it as monolingual as well.
```

File: docs/formats.rst (hunk 15 above)
Comment (nijel) [reply docs/formats.rst:111]: Intersphinx works only between Sphinx documentations, what http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html definitely is not.

File: docs/formats.rst [hunk 17]
```diff
@@ -735,13 +734,13 @@ Example file:
 .. versionadded:: 2.3
 
 .NET Resource (.resx) file is a monolingual XML file format used in Microsoft
```
Comment (comradekingu) [inline docs/formats.rst:736]:
```suggestion
A .NET Resource (.resx) file employs a monolingual XML file format used in Microsoft
```

File: docs/formats.rst [hunk 18]
```diff
@@ -734,14 +733,14 @@ Example file:
 
 .. versionadded:: 2.3
 
-.NET Resource (.resx) file is a monolingual XML file format used in Microsoft
-.NET Applications. It works with .resw files as well as they use identical
+A .NET Resource (.resx) file employs a monolingual XML file format used in Microsoft
+.NET applications. It works with .resw files as well as when they use identical
```
Comment (comradekingu) [inline docs/formats.rst:737]:
```suggestion
.NET applications. It is interchangeable with .resw, when using identical
```

File: docs/formats.rst (hunk 9 above)
Comment (comradekingu) [reply docs/formats.rst]: Was just the wording that had me diddled.

File: docs/formats.rst (hunk 7 above)
Comment (orangesunny) [reply docs/formats.rst]: already fixed this few days ago and forgot to resovle this convesation. Sorry.

File: docs/formats.rst (hunk 8 above)
Comment (orangesunny) [reply docs/formats.rst]: renamed and resolved

File: docs/formats.rst (hunk 12 above)
Comment (orangesunny) [reply docs/formats.rst]: link added

File: docs/formats.rst (hunk 15 above)
Comment (orangesunny) [reply docs/formats.rst:115]: no and resolved

File: docs/formats.rst (hunk 1 above)
Comment (orangesunny) [reply docs/formats.rst]: the

**Pull request merged**